geps/gep-1651.md

```diff
@@ -0,0 +1,298 @@
+signal the default routability using the Gateway's `status.addresses`. See 'Status Addresses`
+for more details.
+
+### Mutability
+Implementations MAY prevent end-users from updating the `routability` value of a Gateway. If
+updates are allowed the semantics and behaviour will depend on the underlying implementation.
+
+### Go
+
+```go
+
+type GatewayRoutability string
+
+const (
+  GatewayRoutabilityPublic  GatewayRoutability  = "Public"
+  GatewayRoutabilityPrivate GatewayRoutability  = "Private"
+  GatewayRoutabilityCluster GatewayRoutability  = "Cluster"
+)
+
+type GatewaySpec struct {
+  // Infrastructure defines infrastructure level attributes about this Gateway instance.
+  Infrastructure GatewayInfrastructure `json:"infrastructure"`
+  // ...
+}
+type GatewayInfrastructure struct {
+  // AddressScope allows the Gateway to specify the accessibility of it's addresseses
+  Routability GatewayRoutability `json:"routability"`
+}
+
+type GatewayAddress struct {
+  // Routability specifies the routable bounds of this address
+  // Predefined values are - Private, Public, Cluster
+  //
+  // This only appears on the addresses in the Gateway's status
+  Routability GatewayRoutability `json:"routability"`
+  // ...
+}
+```
+
+### YAML
+```yaml
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: Gateway
+metadata:
+  name: prod-web
+spec:
+  gatewayClassName: acme-lb
+  infrastructure:
+    routability: Public
+  listeners:
+  - protocol: HTTP
+    port: 80
+```
+
+### Semantics
+
+#### Interaction with GatewayClass
+
+An infrastructure provider may provide a pre-defined set of GatewayClasses that limit the
+routability choices of a Gateway. If the desired Gateway routability is incompatible with the
+GatewayClass it MUST set the condition `Accepted` to `False` with `Reason` set to `Invalid`.
+
+#### Unsupported routability & address values
+
+If a Gateway is unable provide an address for the desired routability it MUST set the condition `Accepted` to `False` with `Reason` set to `Invalid`
+
+#### Status.Addresses
+
+If a Gateway supports the desired 'routability' implementations MUST populate the `status.addresses` with
+an address that has the same routable semantics. The GatewayAddress field `routability` MUST be populated.
+
+Implementations MAY add additional addresses in status, but they MUST be semantically less than the scope
+of the requested scope. For example if a user requests a `Cluster` routable Gateway then the list of addresses
+MUST NOT have a routability of `Public` or `Private`.
+
+## Examples
+
+#### 1. Request a GatewayAddress that is routable within the same cluster
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: Gateway
+metadata:
+  name: prod-web
+spec:
+  gatewayClassName: acme-lb
+  infrastructure:
+    routability: Cluster
+  listeners:
+  - protocol: HTTP
+    port: 80
+```
+#### 2. Request a GatewayAddress with a specific routability and address
```

Comment:
Tiny nit: It seems like some empty lines before these headings would be helpful, everything feels pretty squished together in this section right now.

Reply:
done